crates/nu-command/src/filesystem/ls.rs

```diff
@@ -63,20 +63,30 @@ impl Command for Ls {
         let short_names = call.has_flag("short-names");
 
         let call_span = call.head;
+        let mut new_prefix = false;
+        let mut prefx = vec![];
```

Comment:
This one confuses me a little bit. When we do `ls ..` we should display the files' names as 
`../filenames`. Not exactly sure how do I do that with `PathBuf` and `push`. I tried something, but I think we still need to use `format`? What do you think?

Reply:
@onthebridgetonowhere The `PathBuf::push()` automatically inserts the path separators. So instead of `prefix.push(format!("..{}", MAIN_SEPARATOR))` you can do just `prefix.push("..")`. Later, you can join the filename with just `prefix.push(p)`. 

It's just a detail that simplifies the code a bit. I wrote the comment some time ago when you asked about how to make it less ugly but forgot to "submit" so you couldn't see it :-D.

Reply:
Thanks, I did not know `push` adds the path separators -- though it now obviously makes sense. I am not sure why, but it seems to panic if I do `ls ../../` and I have the code 

`match path.file_name() {
                                    Some(p) => {
                                        prefix_str.push(p);
                                        prefix_str.to_string_lossy().to_string()`
                                        
It suggests that this line has something to do with it: `                let cwd = current_dir(engine_state, stack)?;
`
Need to investigate closer what's going on.

Reply:
@onthebridgetonowhere  I tried the PR and it seems to work fine with the `format!` method so it's mostly just cosmetics. I'll merge this as it is and open a new PR since I noticed the current directory somehow got broken in `ls` (not you PR's fault!). I can try the pathbuf pushing while I'm at it as well.